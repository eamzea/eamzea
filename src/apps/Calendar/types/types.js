export const types = {
  uiOpenModal: "[UI] Open Modal",
  uiCloseModal: "[UI] Close Modal",

  eventSetActive: "[Event] Set Active",
  eventStartAddNew: "[Event] Start Add New",
  eventAddNew: "[Event] Add New",
  eventClearActiveEvent: "[Event] Clear Active Event",
  eventUpdated: "[Event] Updated Event",
  eventDeleted: "[Event] Deleted Event",
  eventLoaded: "[Event] Events Loaded",
  eventLogout: "[Event] Clean Events",

  authCheckingFinish: "[Auth] Finish Checking Login State",
  authStartLogin: "[Auth] Start Login",
  authLogin: "[Auth] Login",
  authStartRegister: "[Auth] Start Register",
  authStartTokenRenew: "[Auth] Start Token Renew",
  authLogout: "[Auth] Logout",
};
