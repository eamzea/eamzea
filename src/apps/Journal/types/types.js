export const types = {
  login: "[Auth] Login",
  logout: "[Auth] Logout",
  uiSetError: "[UI] Set error",
  uiRemoveError: "[UI] Remove error",
  uiStartLoading: "[UI] Start Loading",
  uiFinishLoading: "[UI] Finish Loading",
  notesAddNew: "[Notes] New Note",
  notesNoteActive: "[Notes] Set Active Note",
  notesLoad: "[Notes] Load Notes",
  notesUpdated: "[Notes] Update Note",
  notesFileUrl: "[Notes] Update Image Url",
  notesDelete: "[Notes] Delete Note",
  notesLogoutCleaning: "[Notes] Logout Cleaning",
};
