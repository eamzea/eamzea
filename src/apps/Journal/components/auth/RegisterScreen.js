import React from 'react';
import { Link } from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import validator from 'validator';
import { useDispatch, useSelector } from 'react-redux';
import { setError, removeError } from '../../actions/ui';
import { startRegisterEmailPasswordName } from '../../actions/auth';

const RegisterScreen = () => {
  const dispatch = useDispatch();

  const { msgError } = useSelector(state => state.ui);

  const [formValues, handleInputChange] = useForm({
    name: 'Edgar',
    email: 'eamzea@gmail.com',
    password: 'cesarZEA210394',
    password2: 'cesarZEA210394',
  });

  const { name, email, password, password2 } = formValues;

  const handleRegister = e => {
    e.preventDefault();

    if (isFormValid()) {
      dispatch(startRegisterEmailPasswordName(email, password, name));
    }
  };

  const isFormValid = () => {
    if (name.trim().length === 0) {
      dispatch(setError('Name is Required'));
      return false;
    } else if (!validator.isEmail(email)) {
      dispatch(setError('Email is not valid'));
      return false;
    } else if (password.length <= 5) {
      dispatch(setError('Password should be at least 6 characters'));
      return false;
    } else if (password !== password2) {
      dispatch(setError(`Passwords don't match`));
      return false;
    }

    dispatch(removeError());

    return true;
  };

  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form
        onSubmit={handleRegister}
        className="animate__animated animate__fadeIn animate__faster"
      >
        {msgError && <div className="auth__alert-error">{msgError}</div>}

        <input
          className="auth__input"
          type="text"
          placeholder="Name"
          name="name"
          autoComplete="off"
          value={name}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="text"
          placeholder="Email"
          name="email"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="password"
          placeholder="Password"
          name="password"
          value={password}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="password"
          placeholder="Confirm password"
          name="password2"
          value={password2}
          onChange={handleInputChange}
        />
        <button type="submit" className="btn btn-primary btn-block mb-5">
          Register
        </button>

        <Link to="/journal/auth/login" className="link">
          Already registered?
        </Link>
      </form>
    </>
  );
};

export default RegisterScreen;
