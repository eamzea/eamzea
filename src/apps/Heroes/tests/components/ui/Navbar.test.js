import "@testing-library/jest-dom";
import React from "react";
import { mount } from "enzyme";
import { AuthContext } from "../../../auth/AuthContext";
import { MemoryRouter, Router } from "react-router-dom";
import NavbarHeroes from "../../../components/ui/Navbar";
import { types } from "../../../types/types";

describe("Pruebas en <Navbar/>", () => {
  const historyMock = {
    push: jest.fn(),
    replace: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn(),
  };

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: "Edgar",
    },
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter>
        <Router history={historyMock}>
          <NavbarHeroes />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").text().trim()).toBe("Edgar");
  });

  test("Debe de hacer logout y history", () => {
    wrapper.find("Button").prop("onClick")();

    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.logout,
    });

    expect(historyMock.replace).toHaveBeenCalledWith("/login");
  });
});
