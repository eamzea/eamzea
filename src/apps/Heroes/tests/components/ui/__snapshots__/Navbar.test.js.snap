// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar/> Debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <span
    className="navbar-brand"
  >
    <a
      className="navbar-brand"
      onClick={[Function]}
    >
      Asociaciones
    </a>
  </span>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <p
        className="nav-item nav-link m-0"
      >
        Edgar
      </p>
      <button
        className="nav-item nav-link btn btn-link"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Logout
      </button>
    </div>
  </div>
</nav>
`;
